DPG/Tasks/qaEventTrack.cxx

```diff
@@ -87,16 +87,16 @@ struct qaEventTrack {
   Filter trackFilter = (trackSelection.node() == 0) ||
                        ((trackSelection.node() == 1) && requireGlobalTrackInFilter()) ||
                        ((trackSelection.node() == 2) && requireGlobalTrackWoPtEtaInFilter()) ||
-                       ((trackSelection.node() == 3) && requireTrackCutInFilter(TrackSelectionFlags::kQualityTracks) && requireTrackCutInFilter(TrackSelectionFlags::kInAcceptanceTracks)) ||
-                       ((trackSelection.node() == 4) && requireTrackCutInFilter(TrackSelectionFlags::kQualityTracks)) ||
+                       ((trackSelection.node() == 3) && requireGlobalTrackWoDCAInFilter()) ||
+                       ((trackSelection.node() == 4) && requireQualityTracksInFilter()) ||
                        ((trackSelection.node() == 5) && requireTrackCutInFilter(TrackSelectionFlags::kInAcceptanceTracks));
 
   using TrackIUTable = soa::Join<aod::TracksIU, aod::TrackSelection>;
   Partition<TrackIUTable> tracksIUFiltered = (trackSelection.node() == 0) ||
                                              ((trackSelection.node() == 1) && requireGlobalTrackInFilter()) ||
                                              ((trackSelection.node() == 2) && requireGlobalTrackWoPtEtaInFilter()) ||
```

Comment:
I'm not sure I get it, the track selection is done based on the propagated tracks and not the IU

Reply:
Ok wait maybe I was confusing, indeed. The `TrackSelection` workflow selects on tracks propagated to the primary vertex and seects them, then is creates a bitmap that tells us for each of these tracks which cuts are passed or not. Then here you are filtering the `tracksIU` according to these cuts, namely those of the track itself not at the IU, but at the PV. Is this the logic, because you want to look at exactly the same tracks but looking at their IU version?

Reply:
I'm filtering `tracksIU` based on the decision of propagated tracks so that we have the same tracks that we use in the analysis but at the IU stage, yes

Reply:
perfect, thank you